product-service/src/functions/getProductById/handler.ts

```diff
@@ -4,24 +4,28 @@ import { APIGatewayProxyHandler } from 'aws-lambda';
 
 import { formatJSONResponse } from '@libs/apiGateway';
 import { middyfy } from '@libs/lambda';
-import { ProductsService } from 'src/services/products';
 import { STATUS_CODES } from '@libs/const';
 
+import { getProductById as getProductByIdFromDb } from 'src/services/db';
+
 export const getProductById: APIGatewayProxyHandler = async (event) => {
+    console.log('getProductById lambda is executing', {
+        method: event.httpMethod,
+        pathParameters: event.pathParameters,
+        body: event.body
+    });
+
     try {
         const { productId = '' } = event.pathParameters;
-        const service = new ProductsService();
-        const product = await service.getProductById(productId);
+        const product = await getProductByIdFromDb(productId);
         if (product) {
             return formatJSONResponse(product);
         }
         return formatJSONResponse({
             body: 'Product not found'
         }, STATUS_CODES.NOT_FOUND);
     } catch (error) {
-        return formatJSONResponse({
-            body: `Something bad has happened: ${JSON.stringify(error)}`
-        }, STATUS_CODES.BAD_REQUEST);
+        throw error;
```

Comment:
don't really see a reason to just catch and rethrow an eror. did you forget to add logging here?

Reply:
there is an onError middleware added by middyfy that catch such errors

Reply:
I mean that if you weren't try-catching the error at all, the result would be the same